ChatRPG/API/OpenAiLlmClient.cs

```diff
@@ -1,101 +1,62 @@
 using Microsoft.IdentityModel.Tokens;
-using RestSharp;
-using RestSharp.Authenticators;
+using OpenAI_API;
+using OpenAI_API.Chat;
 
 namespace ChatRPG.API;
 
-public class OpenAiLlmClient : IOpenAiLlmClient, IDisposable
+public class OpenAiLlmClient : IOpenAiLlmClient
 {
-    private const string OpenAiBaseUrl = "https://api.openai.com/v1/";
     private const string Model = "gpt-3.5-turbo";
     private const double Temperature = 0.7;
-    private const double PromptToken1KCost = 0.0015;
-    private const double CompletionToken1KCost = 0.002;
 
-    private readonly ILogger<OpenAiLlmClient> _logger;
-    private readonly RestClient _client;
+    private readonly IHttpClientFactory _httpClientFactory;
+    private readonly OpenAIAPI _openAiApi;
 
-    public OpenAiLlmClient(ILogger<OpenAiLlmClient> logger, IConfiguration configuration,
-        HttpMessageHandler httpMessageHandler)
+    public OpenAiLlmClient(IConfiguration configuration, IHttpClientFactory httpClientFactory)
     {
-        _logger = logger;
+        _httpClientFactory = httpClientFactory;
+        _openAiApi = new OpenAIAPI(configuration.GetSection("ApiKeys").GetValue<string>("OpenAI") ?? string.Empty);
+        _openAiApi.Chat.DefaultChatRequestArgs.Model = Model;
+        _openAiApi.Chat.DefaultChatRequestArgs.Temperature = Temperature;
+    }
 
-        var options = new RestClientOptions(OpenAiBaseUrl)
-        {
-            Authenticator = new JwtAuthenticator(configuration.GetSection("ApiKeys").GetValue<string>("OpenAI") ?? string.Empty),
-            FailOnDeserializationError = false,
-            ConfigureMessageHandler = _ => httpMessageHandler
-        };
-        _client = new RestClient(options);
+    public async Task<string> GetChatCompletion(OpenAiGptMessage input)
+    {
+        var inputList = new List<OpenAiGptMessage> { input };
```

Comment:
`var` - Sarmi kommer efter din booty! 👀

Reply:
```suggestion
        List<OpenAiGptMessage> inputList = new() { input };
```

Reply:
Jeg troede at vi ikke havde et problem med var hvis typen var på højre side af assignment? 😇

Reply:
Haha, det er også rigtigt at det tit er lige så let at læse. Synes bare vi blev engie om at vi gjorde det samme igennem hele projektet for at være consistant. Dvs. ingen `var` in this house. Sarmi finder shotgunen frem og nakker dem ude bagved, så hvis man elsker sine børn, skal man selv redde dem :eyes:

Reply:
Nå okay fair nok, så nakker jeg dem selv, han skal ikke have the pleasure

Reply:
donezo

Reply:
Hold up dont put the blame on me. Synes det er ok med var når typen er på højre side!